great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -129,10 +129,10 @@ def _add_batch_identifiers(
                     f"""RuntimeDataConnector "{self.name}" requires batch_identifiers to be configured, either at the DataConnector or Asset-level."""
                 )
             if batch_identifiers:
-                # TODO: <WILL> discuss whether DataConnector-level batch_identifiers need to be fully deprecated. Commenting for now.
-                # warnings.warn(
-                #    "Specifying batch_identifiers as part of RuntimeDataConnector config is deprecated as of v0.15.01 and will be removed by v0.17. Please configure Assets instead.",
-                # )
+                warnings.warn(
+                    "Specifying batch_identifiers as part of the RuntimeDataConnector config is deprecated as of v0.15.01 and will be removed by v0.18. Please configure batch_identifiers as part of Assets instead.",
+                    DeprecationWarning,
+                )
```

Comment:
`DeprecationWarning` for DataConnector-level `batch_identifiers` has been re-introduced